woocommerce/woocommerce-android
# [Free Trial] Wait for site to sync before showing result to user.

Closes: #8740


### Description

This PR aligns post-installation behavior. It applies "mitigation strategy" introduced in iOS. See this PR https://github.com/woocommerce/woocommerce-ios/pull/9340 and internal communication: pe5pgL-2Cd-p2.

This PR aligns other behaviors to sync with iOS:
- in case of time-out, we won't send the `SITE_CREATION_FAILED` event. (source: https://github.com/woocommerce/woocommerce-ios/pull/9417#issuecomment-1503273303)
- "is jetpack installed" now contains checking two properties: `isJetpackInstalled` and `isJetpackConnected`. Previously we checked only `isJetpackInstalled`

### Testing instructions

Create a new Free Trial site and verify that everything worked as expected.

Other cases are covered with unit-tests and difficult to reproduce (networking). They're covered with unit tests though - both interface adapters and use case layers.
